### PR TITLE
treewide: 6.1: fix compilation error

### PR DESCRIPTION
**Pull request by Ansuel**
Fix multiple compilation error detected while testing some target with kernel 6.1

### CODE REVIEW

File: target/linux/ath79/files/drivers/mfd/rb4xx-cpld.c [hunk 1]
```diff
@@ -151,10 +151,17 @@ static int rb4xx_cpld_probe(struct spi_device *spi)
 				    NULL, 0, NULL);
 }
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(5,17,0)
+static void rb4xx_cpld_remove(struct spi_device *spi)
```
Comment (robimarko) [inline target/linux/ath79/files/drivers/mfd/rb4xx-cpld.c:155]: Why not just drop the remove method completely as it does nothing anyway?

Comment (Ansuel) [reply target/linux/ath79/files/drivers/mfd/rb4xx-cpld.c:155]: Just checked the driver, yes can be done.

File: target/linux/generic/hack-6.1/902-debloat_proc.patch [hunk 2]
```diff
@@ -406,3 +406,16 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
...
  }
  
+--- a/net/ipv4/inet_timewait_sock.c
++++ b/net/ipv4/inet_timewait_sock.c
+@@ -272,6 +272,9 @@ static struct pernet_operations ip_rt_pr
+ 	 */
+ 
+ 	if (!rearm) {
+-		bool kill = timeo <= 4*HZ;
++#ifndef CONFIG_PROC_STRIPPED
```
Comment (robimarko) [inline target/linux/generic/hack-6.1/902-debloat_proc.patch:416]: Is this upstream code or?

Comment (Ansuel) [reply target/linux/generic/hack-6.1/902-debloat_proc.patch:416]: hack patch used by tiny target

Comment (robimarko) [reply target/linux/generic/hack-6.1/902-debloat_proc.patch:416]: Is there a measurement on how much space does this `/proc` downstream patch even save?

**Comment by xback** [comment]
https://git.openwrt.org/?p=openwrt/staging/xback.git;a=commit;h=b537a0d75584e19dabc98ba9c2a97ea5033b5364

Feel free to use it :)

**Comment by Ansuel** [comment]
@xback sure let me cherry-pick!

File: target/linux/generic/hack-6.1/902-debloat_proc.patch [hunk 3]
```diff
@@ -406,3 +406,16 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
...
+--- a/net/ipv4/inet_timewait_sock.c
++++ b/net/ipv4/inet_timewait_sock.c
+@@ -272,7 +272,9 @@ static struct pernet_operations ip_rt_pr
+ 	 */
+ 
+ 	if (!rearm) {
+-		bool kill = timeo <= 4*HZ;
++#ifndef CONFIG_PROC_STRIPPED
++		bool kill = timeo <= 4*HZ;
++#endif
```
Comment (KanjiMonster) [inline target/linux/generic/hack-6.1/902-debloat_proc.patch:418]:
```suggestion
@@ -272,7 +272,7 @@ static struct pernet_operations ip_rt_pr
 	 */
 
 	if (!rearm) {
-		bool kill = timeo <= 4*HZ;
+		bool  __maybe_unused kill = timeo <= 4*HZ;

```

should also do the trick with a bit less change and without if-deffery.

Comment (Ansuel) [reply target/linux/generic/hack-6.1/902-debloat_proc.patch]: Thanks! Nice idea

**Pull request merged**